src/Monitoring/MetricsAgent.php

```diff
@@ -12,5 +12,43 @@
 
 interface MetricsAgent
 {
+    /**
+     * For recording elapsed times of execution
+     *
+     * @param MetricId          $metric
+     * @param DateTimeImmutable $start_time
+     * @param DateTimeImmutable $end_time
+     */
     public function addTimer(MetricId $metric, DateTimeImmutable $start_time, DateTimeImmutable $end_time): void;
+
+    /**
+     * Similar to a timer but for recording values which are not time based but where min, max, mean are useful
+     *
+     * @param MetricId $metric
+     * @param float    $value
+     */
+    public function addSample(MetricId $metric, float $value): void;
+
+    /**
+     * Simple counter with single positive increment
+     *
+     * @param MetricId $metric
+     */
+    public function incrementCounterByOne(MetricId $metric): void;
+
+    /**
+     * Similar to a counter integral / key value but on most agents only the most recent value is recorded
+     *
+     * @param MetricId $metric
+     * @param float    $value
+     */
+    public function setGauge(MetricId $metric, float $value): void;
+
+    /**
+     * A simple key value store
```

Comment:
That kinda implies there's no relationship between datapoints?

Maybe something more along the lines

```
/** 
 * Record arbitrary point-in-time values - e.g. a cache hit rate - similar to a key-value store, on most agents only the most recent value is recorded.
 */
setGauge()

/**
 * Record a counter value where you have the current value and want the agent / backend to calculate the differential between each measurement.
 * 
 * For example, `total connections since last restart`.
 */
 recordCounterIntegral()
```